### ISSUE TITLE
Streams seem to disappear when clusters are connected via gateway

### ISSUE THREAD

**Issue by pcsegal** [nats-io/nats-server#4816]
### Observed behavior

I'm trying to set up a gateway connection between two clusters, A and B, in order to migrate streams and KV buckets from cluster A to cluster B and decommission cluster B.

To create the gateway connections, I'm using an arbiter cluster with a single node.

I have added a reproducible example below, including NATS configuration files and a script, `test.sh`, that runs the example and reproduces the issue I'm seeing.

The steps in `test.sh` do the following:

1. Start cluster A and cluster B (they define a gateway block with listening port, but are not connected; they are independent clusters).
2. Create a KV bucket in cluster A with some data.
3. Start the arbiter. This connects cluster A and cluster B into a supercluster.
4. Check that the KV bucket is visible from both cluster A and cluster B.
5. Edit the KV bucket's stream to move it to cluster B.
6. For each node in cluster A, shut it down and run `nats server raft peer-remove` to decommission it.
7. Verify that the KV bucket is in cluster B.

There is a problem that I see in step 3 after repeatedly running `test.sh` a few times. After  the arbiter is up, the KV bucket sometimes disappears, and is no longer visible in any cluster. This happens randomly when re-running `test.sh`. Note I used KV buckets here just as an example; this happens when I test it with pure streams as well.

Here is an example of what the output looks like when this happens; this is printed after starting up the arbiter and waiting for a few seconds:

```
KV buckets in cluster-a:
No Key-Value buckets found
KV buckets in cluster-b:
No Key-Value buckets found
nats: error: could not pick a Stream to operate on: unknown stream "KV_example"
```

### Expected behavior

I expect that, after the arbiter is up and the gateway connections are established, any stream that previously existed in cluster A or B continues existing in cluster A or B.

### Server and client version

NATS server version: 2.10.5.
NATS client version: 0.1.1

### Host environment

Ubuntu 20.04.6.

### Steps to reproduce

```
#!/bin/bash

set -euo pipefail

# Clean up: make sure there are no NATS servers left over from a previous run of this script.

pkill -f -9 'nats.*cluster-a' || true
pkill -f -9 'nats.*cluster-b' || true
pkill -f -9 'nats.*arbiter' || true

# Clean up data directories used in this example to make sure we start from scratch.

rm -rf /tmp/nats-gateway-issue || true

mkdir -p /tmp/nats-gateway-issue

# Start cluster-a.

nats-server -c cluster-a-0.conf > /tmp/nats-gateway-issue/cluster-a-0.log 2>&1 &
nats-server -c cluster-a-1.conf > /tmp/nats-gateway-issue/cluster-a-1.log 2>&1 &
nats-server -c cluster-a-2.conf > /tmp/nats-gateway-issue/cluster-a-2.log 2>&1 &

# Start cluster-b.

nats-server -c cluster-b-0.conf > /tmp/nats-gateway-issue/cluster-b-0.log 2>&1 &
nats-server -c cluster-b-1.conf > /tmp/nats-gateway-issue/cluster-b-1.log 2>&1 &
nats-server -c cluster-b-2.conf > /tmp/nats-gateway-issue/cluster-b-2.log 2>&1 &

sleep 2

# Wait for cluster-a and cluster-b to be ready.

curl --fail --silent --retry 5 --retry-delay 1 http://localhost:8222/healthz > /dev/null
curl --fail --silent --retry 5 --retry-delay 1 http://localhost:8223/healthz > /dev/null
curl --fail --silent --retry 5 --retry-delay 1 http://localhost:8224/healthz > /dev/null
curl --fail --silent --retry 5 --retry-delay 1 http://localhost:8225/healthz > /dev/null
curl --fail --silent --retry 5 --retry-delay 1 http://localhost:8226/healthz > /dev/null
curl --fail --silent --retry 5 --retry-delay 1 http://localhost:8227/healthz > /dev/null

sleep 3

# Add a KV bucket with some data to cluster-a.

nats -s localhost:4222 kv add example

nats -s localhost:4222 kv put example key1 value1
nats -s localhost:4222 kv put example key2 value2
nats -s localhost:4222 kv put example key3 value3

# Start arbiter.

nats-server -c arbiter.conf > /tmp/nats-gateway-issue/arbiter.log 2>&1 &
sleep 1
curl --fail --silent --retry 5 --retry-delay 1 http://localhost:8228/healthz > /dev/null

# Arbiter up. Now cluster-a and cluster-b plus arbiter should form a supercluster.

sleep 3

# Both clusters should see the KV bucket.

echo "KV buckets in cluster-a:"
nats -s localhost:4222 kv ls #--user a --password a
echo "KV buckets in cluster-b:"
nats -s localhost:4225 kv ls #--user a --password a

# Migrate KV bucket's stream to cluster-b.

nats -s localhost:4222 stream edit KV_example --cluster cluster-b --force

sleep 5

# Kill cluster-a.

pkill -f "cluster-a-0"
sleep 1
nats -s localhost:4225 server raft peer-remove cluster-a-0 --user admin --password pwd

pkill -f "cluster-a-1"
sleep 1
nats -s localhost:4225 server raft peer-remove cluster-a-1 --user admin --password pwd

pkill -f "cluster-a-2"
sleep 1
nats -s localhost:4225 server raft peer-remove cluster-a-2 --user admin --password pwd

# KV bucket should be in cluster-b.

nats -s localhost:4225 kv ls
```

**Comment by pcsegal** [comment]
Had forgotten to include the configuration files in the reproducible example. Edited to include it.

**Comment by derekcollison** [comment]
What role is the arbiter playing? Why not connect cluster A directly to cluster B via GWs? Of extend cluster A with new nodes representing the new servers, and have them tagged, and use tags to move the assets?

**Comment by pcsegal** [comment]
True, this is not the simplest way to do a data migration. In fact, I ended up using a simpler strategy.

But, in any case, I posted this issue here because it looks like establishing a gateway connection shouldn't cause this observed behavior.

Outside of this specific migration scenario,  the role of the arbiter would be for JetStream fault tolerance: if one cluster goes down, you still have 2 other clusters, and so JetStream is still available.

**Comment by pcsegal** [comment]
Hi, I'm wondering if you had time to investigate this further.

**Comment by ripienaar** [comment]
Does the latest version still exhibit this behaviour ? There are some RC releases ready also for the next.

**Comment by pcsegal** [comment]
Yes, I tested it again with 2.10.17-RC6 and it still seems to exhibit the behavior of sometimes not showing any KV anymo after starting up the arbiter node.

**Comment by ripienaar** [comment]
This config where the arbiter node is the only one with the cluster shape defines is probably not really a supported configuration, jetstream wants a mostly static setup and wants to know the shape of things, you should list the full cluster everywhere rather than have this one arbitrer that configures up the cluster, it's not just not a supported approach I think.

It might work now and then and in some cases but its just not going to survive any outages or situations where the arbiter isnt around and the arbiter becomes a quite nasty SPOF.

Better use the software as designed.

**Comment by arkh-consensys** [comment]
I have observed a similar behaviour by only adding system account to jetstream k8s helm configuration and we lost the stream completely.

Adding such config to the helm charts

`      jetstream:
        max_memory_store: << 1GB >>
      accounts: {
          SYS: {
              users: [
                  { user: admin, password: << $ADMIN_PASSWORD >> }
              ]
          },
      }
      system_account: SYS`
      
 Caused our stream to be removed!
 
 I assume it is not safe to add account (or system account) to a running server. 
 Previously our Jetstream server had no accounts enabled.

**Comment by wallyqs** [comment]
@arkh-consensys it had a system account but it was named `$SYS` which is the default one and then here it was renamed to `SYS`.

**Comment by garrett-sutton** [comment]
@pcsegal would you mind sharing the simpler approach that you took to solving this problem? 

I have been running into this problem in my testing. There doesn't seem to be much documentation or record of others trying to do this sort of thing from my searching.

**Comment by pcsegal** [comment]
@garrett-sutton Hi, sorry for the big delay. Since the original problem was data migration to a new cluster, and it was not a situation where downtime was an issue, we ended up using very simple manual backup / restore (https://docs.nats.io/running-a-nats-service/nats_admin/jetstream_admin/disaster_recovery).

**Comment by pcsegal** [comment]
@garrett-sutton Would you describing the configuration you were using when you ran into this problem? Was it similar to the "arbiter" approach that I described above?
